### PR TITLE
fix: Simplify / update build instructions.

### PR DESCRIPTION
**Pull request by lizroth**
*Issue #, if available:*

*Description of changes:* Update build instructions. Reference the AWS SDK for C++ build instructions where possible to deduplicate setup. Simplify the steps for the `aws-c-common` branch.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Review by alex-chew — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
 
 [Security issue notifications](./CONTRIBUTING.md#security-issue-notifications)
 
-## Dependencies
+## Build recipes with KMS support
```
Comment (juneb) [inline README.md:12]: Can we be very clear here? "Building the AWS Encryption SDK for C (with support for AWS KMS)"

(The current statement can be interpreted as using KMS to help with the build.)

File: README.md [hunk 2]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
 
-The only direct dependencies of this code are OpenSSL 1.0.2 or higher or 1.1.0 or higher and
-[aws-c-common](https://github.com/awslabs/aws-c-common) v0.4.42. You will also need
-a C compiler and CMake 3.9 or higher.
+We will demonstrate some simple build recipes for Linux, Mac, and Windows operating systems. These 
+recipes assume a fresh system with default installs of dependency packages.
```
Comment (juneb) [inline README.md:15]: Let's define "fresh". Also s/installs/installations. 

I love the voice, really, but many of our customers are global. Let's be clear for them.

File: README.md [hunk 3]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
 
-To integrate with [KMS](https://aws.amazon.com/kms/) the AWS Encryption SDK for C also requires
-the [AWS SDK for C++](https://github.com/aws/aws-sdk-cpp), a C++ compiler, and libcurl.
+The Linux and Mac recipes install everything in the standard directories in /usr/local. The
+Windows recipe installs everything in an install directory placed at the directory you are in
+when you start the build process. To change the installation directory, if desired, see the Tips
+and Tricks section at the end of this README.
```
Comment (juneb) [inline README.md:20]:
```suggestion
The Windows instructions install everything in your current directory (where you run the build process). To change the installation directory, see the Tips and Tricks section at the end of this README.
```

File: README.md [hunk 4]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
+
+The AWS Encryption SDK for C can be used with AWS KMS, but it is not required. If you want to build
+a minimal version of the ESDK without KMS support, see "Minimal C build without KMS support", below.
```
Comment (juneb) [inline README.md:23]: s/KMS/AWS KMS throughout, please.

File: README.md [hunk 5]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
-using the cmake argument `-DBUILD_SHARED_LIBS=ON`. You can change them to static library
-builds by just changing `ON` to `OFF`.
-
-Once you have built each dependency, install it so it can be picked up by the next build.
-
 ### If you are working on an EC2 instance, regardless of operating system
 
 For best results, create an EC2 instance with an instance profile that at a
 minimum has KMS permissions for Encrypt, Decrypt, and GenerateDataKey for
 at least one KMS CMK in your account. You will not need any other AWS
```
Comment (juneb) [inline README.md:36]:
```suggestion
For best results, create an EC2 instance with an instance profile that at includes AWS KMS permissions for Encrypt, Decrypt, and GenerateDataKey on
at least one KMS CMK in your account. You will not need any other AWS
```

File: README.md [hunk 6]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
+1. OpenSSL 1.0.2 or newer, or 1.1.0 or newer
+1. CMake 3.9 or newer
+1. C/C++ compiler
+1. aws-c-common, typically bundled with the AWS SDK for C++
+1. The AWS SDK for C++ version 1.9.35 or newer
 
-#### Ubuntu: (Option 1) C and C++ build dependencies
+The AWS SDK for C++ and the AWS Encryption SDK for C share dependencies on OpenSSL, aws-c-common, and CMake, and 
+the AWS SDK for C++ has some additional dependencies and prerequisites of its own. See [AWS SDK for
+C++: Getting Started](https://github.com/aws/aws-sdk-cpp#getting-started).
```
Comment (juneb) [inline README.md:49]:
```suggestion
The AWS SDK for C++ and the AWS Encryption SDK for C share dependencies on OpenSSL, aws-c-common, and CMake. The AWS SDK for C++ has additional dependencies and prerequisites. See [AWS SDK for
C++: Getting Started](https://github.com/aws/aws-sdk-cpp#getting-started).
```

File: README.md [hunk 7]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
-aws-c-common for you.
-
-Do a KMS-only build of the AWS SDK for C++. If you want to use the AWS SDK for C++ for
-other AWS services, you can omit the `-DBUILD_ONLY="kms"` argument, but the build will take much longer.
-
-    git clone https://github.com/aws/aws-sdk-cpp.git
-    mkdir build-aws-sdk-cpp && cd build-aws-sdk-cpp
-    cmake -DBUILD_SHARED_LIBS=ON -DBUILD_ONLY="kms" -DENABLE_UNITY_BUILD=ON ../aws-sdk-cpp
-    make && sudo make install ; cd ..
+You need to compile both the AWS Encryption SDK for C and its dependencies as either all
```
Comment (juneb) [inline README.md:51]: Delete "both." There are more than two things.

File: README.md [hunk 8]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
+shared or all static libraries. These instructions will use all static library builds. Static library
+builds are the default for the AWS SDK for C++ and aws-c-common.
 
-Now skip to the "Ubuntu: Build and install the AWS Encryption SDK for C" section below.
+If you would like to build shared libraries, you will need to supply the `-DBUILD_SHARED_LIBS=ON` flag to build
```
Comment (juneb) [inline README.md:55]:
```suggestion
To build shared libraries, specify the `-DBUILD_SHARED_LIBS=ON` flag to build
```

File: README.md [hunk 9]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
+aws-c-common, the AWS SDK for C++, and the AWS Encryption SDK.
 
-#### Ubuntu: (Option 2) C only build dependencies
+### Linux Build Recipe
 
-    sudo apt-get update
-    sudo apt-get install -y libssl-dev cmake gcc
+First you will need to build the AWS SDK for C++. That will install the shared dependencies.
```
Comment (juneb) [inline README.md:60]:
```suggestion
First, build the AWS SDK for C++. That will install the shared dependencies.
```

File: README.md [hunk 10]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
 
-Now build and install aws-c-common:
+If you only need AWS SDK for C++ to use the AWS Encryption SDK, you have the option to build only the AWS KMS SDK.
+Add the `-DBUILD_ONLY="kms"` flag to `cmake` in the instructions provided.
```
Comment (juneb) [inline README.md:63]:
```suggestion
If you need the AWS SDK for C++ only to use the AWS Encryption SDK, you can build only the AWS KMS SDK.
To do so, add the `-DBUILD_ONLY="kms"` flag to `cmake` as noted in the [AWS SDK for C++ build instructions](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/setup-linux.html).
```

File: README.md [hunk 11]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
 
-    git clone https://github.com/awslabs/aws-c-common.git
-    mkdir build-aws-c-common && cd build-aws-c-common
-    cmake -DBUILD_SHARED_LIBS=ON ../aws-c-common
-    make && sudo make install ; cd ..
+[Follow the AWS SDK for C++ build instructions](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/setup-linux.html).
```
Comment (juneb) [inline README.md:65]: Delete. Moved this into the previous sentence.

File: README.md [hunk 12]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
     mkdir build-aws-encryption-sdk-c && cd build-aws-encryption-sdk-c
-    cmake -DBUILD_SHARED_LIBS=ON ../aws-encryption-sdk-c
+    cmake ../aws-encryption-sdk-c
     make && sudo make install ; cd ..
 
-You have successfully built and installed the AWS Encryption SDK for C.
+### MacOS Build Recipe
 
-### Building on Mac
+[Homebrew](https://brew.sh) is a convenient way to obtain build tools on MacOS systems.
```
Comment (juneb) [inline README.md:76]: s/obtain/get. Also s/on/for (nit).

File: README.md [hunk 13]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
 
-(Note: Installing the AWS SDK for C++ through Homebrew does a full build, which
-takes much longer than the KMS-only build. For these reasons, we recommend doing a source
-build of the AWS SDK for C++ yourself, as we will demonstrate below.)
-
-Start in the directory where you want to do your build.
+The AWS SDK for C++ can be installed with Homebrew, which will install the full AWS SDK. If you 
+only need the AWS SDK for C++ to use the AWS Encryption SDK, you have the option to build only the AWS KMS SDK.
+Follow [these directions](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/setup-linux.html#setup-linux-from-source) 
+and specify the `-DBUILD_ONLY="kms"` flag to `cmake` in the instructions provided.
```
Comment (juneb) [inline README.md:85]:
```suggestion
need the AWS SDK for C++ only to use the AWS Encryption SDK, you can build only the AWS KMS SDK.
See [these directions](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/setup-linux.html#setup-linux-from-source)
and specify the `-DBUILD_ONLY="kms"` flag to `cmake` in the instructions provided.
```

File: README.md [hunk 14]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
     mkdir build-aws-encryption-sdk-c && cd build-aws-encryption-sdk-c
-    cmake -G Xcode -DBUILD_SHARED_LIBS=ON -DOPENSSL_ROOT_DIR="/usr/local/opt/openssl@1.1" ../aws-encryption-sdk-c 
+    cmake -G Xcode -DOPENSSL_ROOT_DIR="/usr/local/opt/openssl@1.1" ../aws-encryption-sdk-c 
     xcodebuild -target install; cd ..
 
-You have successfully built and installed the AWS Encryption SDK for C.
+### Windows Build Recipe
 
-### Building on Windows
+**Note**: See the docker-images folder for some Windows build recipes that automate many of these steps.
```
Comment (juneb) [inline README.md:96]: What docker-images folder? Link? Is there any reason not to use the automation? Should we mark the alternate steps as manual?

File: README.md [hunk 15]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
 
-**Note**: _see the docker-images folder for some Windows build recipes that automate many of theses steps_
+Start by installing Visual Studio version 15 or later with the Windows Universal C Runtime and [Git for Windows](https://git-scm.com/download/win).
```
Comment (juneb) [inline README.md:98]: s/Start by installing/Install

File: README.md [hunk 16]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
 
-Start by installing Visual Studio version 15 or later with the Windows Universal C Runtime and
-[Git for Windows](https://git-scm.com/download/win).
+Use the "x64 Native Tools Command Prompt" for all commands listed here. Run the following commands in the directory where you want to do the build and installation.
```
Comment (juneb) [inline README.md:100]: Why use that command prompt? Is it the only one that works? Or the only one we tested?

File: README.md [hunk 17]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
-directory where you want to do the build and installation.
+Use vcpkg to integrate with Visual Studio and to install prerequisites:
 
     mkdir install && mkdir build && cd build
     git clone https://github.com/Microsoft/vcpkg.git
     cd vcpkg && .\bootstrap-vcpkg.bat
     .\vcpkg install curl:x64-windows openssl:x64-windows && cd ..
 
-#### Windows: (Option 1) C and C++ build dependencies
+You may also want to integrate vcpkg and Visual Studio: `.\vcpkg\vcpkg integrate install`
```
Comment (juneb) [inline README.md:111]: Does the specified command do the integration? If so:

To integrate vcpkg and Visual Studio (optional), use the following command: `.\vcpkg\vcpkg integrate install`

File: README.md [hunk 18]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
 
-Both aws-sdk-cpp and aws-c-common are required, but the installation of aws-sdk-cpp will install
-aws-c-common for you.
+Build the AWS SDK for C++. This installs the aws-c-common dependency too. If you only need AWS SDK for C++ to use the
```
Comment (juneb) [inline README.md:113]:
```suggestion
Build the AWS SDK for C++. This installs the aws-c-common dependency, too. If you need the AWS SDK for C++ only to use the 
```

Place "only" as close to the reference as possible.

File: README.md [hunk 19]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
     cd vcpkg && .\bootstrap-vcpkg.bat
     .\vcpkg install curl:x64-windows openssl:x64-windows && cd ..
 
-#### Windows: (Option 1) C and C++ build dependencies
+You may also want to integrate vcpkg and Visual Studio: `.\vcpkg\vcpkg integrate install`
 
-Both aws-sdk-cpp and aws-c-common are required, but the installation of aws-sdk-cpp will install
-aws-c-common for you.
+Build the AWS SDK for C++. This installs the aws-c-common dependency too. If you only need AWS SDK for C++ to use the 
+AWS Encryption SDK, you have the option to build only the AWS KMS SDK. Remove `-DBUILD_ONLY=kms` to build the entire AWS SDK for C++.
```
Comment (juneb) [inline README.md:114]:
```suggestion
AWS Encryption SDK, you can build only the AWS KMS SDK. To build the entire AWS SDK for C++, delete `-DBUILD_ONLY=kms` from ???.
```

To do A, do B. (for left-to-right languages)

File: README.md [hunk 20]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
-    git clone https://github.com/awslabs/aws-c-common.git
-    mkdir build-aws-c-common && cd build-aws-c-common
-    cmake -DCMAKE_INSTALL_PREFIX=%cd%\..\..\install -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DCMAKE_TOOLCHAIN_FILE=%cd%\..\vcpkg\scripts\buildsystems\vcpkg.cmake -G Ninja ..\aws-c-common
+    git clone --recurse-submodules https://github.com/aws/aws-encryption-sdk-c.git
+    mkdir build-aws-encryption-sdk-c && cd build-aws-encryption-sdk-c
+    cmake -DCMAKE_INSTALL_PREFIX="%cd%\..\..\install" -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DCMAKE_TOOLCHAIN_FILE"=%cd%\..\vcpkg\scripts\buildsystems\vcpkg.cmake" -G Ninja ..\aws-encryption-sdk-c
     cmake --build . && cmake --build . --target install && cd ..
 
-#### Windows: Build and install the AWS Encryption SDK for C
+## Building C only, without KMS support
```
Comment (juneb) [inline README.md:128]: Building the AWS Encryption SDK for C (without support for AWS KMS)

File: README.md [hunk 21]
```diff
@@ -9,247 +9,128 @@ Also, see the [API documentation](https://aws.github.io/aws-encryption-sdk-c/htm
...
 
-    git clone https://github.com/aws/aws-encryption-sdk-c.git
-    mkdir build-aws-encryption-sdk-c && cd build-aws-encryption-sdk-c
-    cmake -DCMAKE_INSTALL_PREFIX=%cd%\..\..\install -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DCMAKE_TOOLCHAIN_FILE=%cd%\..\vcpkg\scripts\buildsystems\vcpkg.cmake -G Ninja ..\aws-encryption-sdk-c
-    cmake --build . && cmake --build . --target install && cd ..
+If you don't need AWS KMS support, you can build the AWS Encryption SDK for C without the AWS SDK. This build is C-only with no C++ support
+required.
 
-You have successfully built and installed the AWS Encryption SDK for C.
+To build without KMS support, follow the build instructions above for your platform, substituting aws-c-common for aws-sdk-cpp.
```
Comment (juneb) [inline README.md:133]: s/KMS/AWS KMS (per kenbeer@)

File: README.md (hunk 17 above)
Comment (lizroth) [reply README.md:108]: This is one of the things that I was unhappy with and am fixing, so hold this thought please for the next round

File: README.md (hunk 11 above)
Comment (lizroth) [reply README.md:63]: I put the instructions in this order with the intent that I state the required step (first, build the SDK); then I explain what modifications the customer must make to the instructions ("if you only need..."); then I finally link to the directions after clarifying what needs to change.

My concern about putting the instructions link first/combined with the previous sentence is that the customer will correctly click the link and go follow those instructions without realizing that our instructions have a couple of notes that apply to those steps.

Does that make sense? Should I still change this, or change it in some different way?

File: README.md (hunk 14 above)
Comment (lizroth) [reply README.md:94]: https://github.com/aws/aws-encryption-sdk-c/tree/master/docker-images

Requiring customers to use Docker in particular creates another dependency to get started working with the SDK. These generic instructions guide customers through bootstrapping their project generically on any system.

I'll add a link

File: README.md (hunk 16 above)
Comment (lizroth) [reply README.md:98]: It's the one to have permissions to install the libraries in the default path and to have all of the developer tools populated on the command line without needing to modify or specify the path

**Review by lizroth — COMMENTED** [review]
Most feedback has been incorporated -- a couple of follow up comments/questions to your feedback so far. Thank you!

**Comment by texastony** [comment]
According to Alex Chew, this PR is awaiting approval from June due to Doc Impact. 

We have an open issue (#717) that is related to this.

Also, we will have to manually rebase this.

**Comment by texastony** [comment]
Errr... #127 may not be related... it appears the requester has successfully built the library, but cannot use the library in any other project... which is interesting...

**Comment by feliperodri** [comment]
cc. @alex-chew

File: README.md (hunk 11 above)
Comment (juneb) [reply README.md:63]: That's fine.

File: README.md (hunk 16 above)
Comment (juneb) [reply README.md:98]: Makes sense.

**Pull request merged**